datapackage_pipelines_measure/processors/add_outputs_resource.py

```diff
@@ -0,0 +1,138 @@
+import re
+import json
+import dateutil
+import urllib
+
+from datapackage_pipelines.generators import slugify
+from datapackage_pipelines.wrapper import ingest, spew
+
+from datapackage_pipelines_measure.processors import google_utils
+
+import logging
+log = logging.getLogger(__name__)
+
+TIMESTAMP_COL = 'A'
+FAR_PAST_START_DATE = '1990-01-01'
+
+
+def _request_data_from_google_spreadsheet(start_date):
+    '''
+    Build a google charts query and append it to an authorised spreadsheets
+    request, returning the response.
+    '''
+    def _build_charts_query(start_date):
+        '''
+        Build and return a charts query to fetch the most recent rows from the
+        spreadsheet, based on the most recent data collected for this source.
+        '''
+        query = '''
+            SELECT *
+            WHERE {timestamp} > date '{start_date}'
+            ORDER BY {timestamp}
+            '''.format(timestamp=TIMESTAMP_COL, start_date=start_date)
+        query = query.strip()
+        return urllib.parse.quote(query)
+
+    def _parse_response_to_dict(response):
+        '''Parse the response from google api and return the bit we want as a
+        native dict'''
+        regexp = re.compile(b"(^\/\*O_o\*\/\\ngoogle\.visualization\.Query\.setResponse\(|\);$)") # noqa
```

Comment:
That's my favourite bit! It's a protection to prevent [content sniffing](https://miki.it/blog/2014/7/8/abusing-jsonp-with-rosetta-flash/).